core/src/main/java/io/fabric8/maven/core/util/PluginServiceFactory.java

```diff
@@ -235,7 +238,8 @@ public int hashCode() {
 
         /** {@inheritDoc} */
         public int compareTo(ServiceEntry o) {
-            return order - o.order;
+            int ret = this.order - o.order;
+            return ret != 0 ? ret : this.className.compareTo(o.className);
```

Comment:
Problem was that `equals()` and `compareTo()` were not consistent and a `TreeSet` always used `compareTo() == 0` for equality.